### DISCUSSION TITLE
Please allow profile component without an avatar

### DISCUSSION THREAD

**Discussion by ryanmortier** [livewire/flux#503] [Feature Requests]
If I'm using the `<flux:profile>` component in a header/navbar/sidebar, I'd like the ability to use it without an avatar image and instead only using the user's name. Example:

![CleanShot 2024-10-22 at 09 40 52](https://github.com/user-attachments/assets/3bdd92af-9229-4e6d-986e-14638fb6a77b)

Right now I'm simply overriding the component by adding a conditional if statement:

```diff
+    <?php if ($avatar): ?>
        <div class="size-8 bg-zinc-400 rounded overflow-hidden">
            <?php if (is_string($avatar)): ?>
                <img src="{{ $avatar }}" />
            <?php else: ?>
                {{ $avatar }}
            <?php endif; ?>
        </div>
+    <?php endif; ?>
```

Can this be added to core?

edit: I understand that I could likely achieve this using a dropdown component or something but then I need to figure out which styles to use and keep these styles updated with Flux updates. It sort of goes against the reasoning for using a project like Flux imo.

**Comment by ju5t** [comment]
> edit: I understand that I could likely achieve this using a dropdown component or something but then I need to figure out which styles to use and keep these styles updated with Flux updates. It sort of goes against the reasoning for using a project like Flux imo.

I don't understand. `flux:profile` is just a button. You don't have to use it. You can use any Flux-button for it.

**Reply by ryanmortier** [reply]
Tried that, it doesn't look the same.

**Reply by ju5t** [reply]
What do you expect it to look like? Because the image you posted can be replicated with a button. 

```blade
<flux:dropdown>
    <flux:button icon-trailing="chevron-down" variant="subtle">Your Name</flux:button>

    <flux:menu>
        <flux:menu.item icon="plus">New post</flux:menu.item>
    </flux:menu>
</flux:dropdown>
```

You can of course us a different variant like `primary` for the button if that's closes to the style you're looking for.

**Reply by ryanmortier** [reply]
They're just off slightly.

For example here is the `<flux:profile>`

![CleanShot 2024-10-22 at 15 20 56](https://github.com/user-attachments/assets/9c20c23e-6ab0-4083-b556-b374d6601162)

Here is a `<flux:button variant="subtle">`

![CleanShot 2024-10-22 at 15 21 05](https://github.com/user-attachments/assets/50da5018-109f-4f71-95c4-5ecfeeff63c4)

Here is a `<flux:button variant="ghost">`

![CleanShot 2024-10-22 at 15 22 03](https://github.com/user-attachments/assets/14d18de7-10c1-47d0-9b47-464acefb63c5)

Here is a `<flux:button variant="primary">`

![CleanShot 2024-10-22 at 15 23 46](https://github.com/user-attachments/assets/89becf80-d5c1-4e1e-8d21-b3faf6aab7bf)

The only one that uses the same color as the navbar links is the `<flux:profile>`.

**Reply by ju5t** [reply]
Although not the cleanest, a workaround would be to add `!text-zinc-500` to the class of the subtle or ghost variant. Another workaround would be to use the `flux:navbar.item` as a button in the dropdown, as it uses `text-zinc-500` already.

I think this needs to be looked at, but I don't think it should be changed in the `flux:profile` component. Because when you remove the avatar it's just a button. Curious what the Flux-team thinks of this.
